### PR TITLE
chore(eslint-config): upgrade config to use ESLint v9 & FlatConfig

### PR DESCRIPTION
**Pull request by grixu**
- Migration to ESLint v9 and adoption of `FlatConfig` for a more modern and flexible configuration approach.
- Each module (`ecma`, `typescript`, `nextjs`, `playwright`, `style`, `architecture`) is a factory for ESLint configurations, which can be used individually or together in projects.
- Enhanced support for ECMAScript, TypeScript, Next.js, Playwright, and architectural rules.
- Integration of Prettier and Perfectionist plugins for improved code formatting and consistency.
- Updated rules and settings to align with the latest best practices and standards.
- Improved handling of import/export rules and JSON file sorting.
- Removal of deprecated and non-compliant plugins.

I've added the `resolutions` field to the root `package.json` to allow local linting in this repo, due to its using the ESLint v8 and Prettier v2. The package itself is linted & formatted using itself.

### CODE REVIEW

File: engineering-toolkit/eslint-config/.prettierrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": false,
+  "printWidth": 120
+}
```
Comment (FRSgit) [inline engineering-toolkit/eslint-config/.prettierrc:4]: Question(optional): Shouldn't eslint-config use prettier-config and vice versa? 😄

File: engineering-toolkit/eslint-config/eslint.config.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { concat } from "eslint-flat-config-utils";
+
+import { ecma, style } from "./src/index.js";
+
+export default concat(ecma(), style());
```
Comment (FRSgit) [inline engineering-toolkit/eslint-config/eslint.config.js:5]: Offtop: does this concat does any more advanced rule concatenation, or only resolves promises and pushes them back into an array?

File: engineering-toolkit/eslint-config/package.json [hunk 3]
```diff
@@ -1,25 +1,17 @@
...
-    "./next-strict": "./src/next-strict.js"
+    ".": "./src/index.js"
   },
   "homepage": "https://github.com/vuestorefront/vue-storefront/engineering-toolkit/eslint-config",
   "contributors": [
     {
-      "name": "Maciej Rybaniec",
-      "email": "maciej.rybaniec@gmail.com"
+      "name": "Mateusz Gostański",
+      "email": "mgostanski@alokai.com"
```
Comment (FRSgit) [inline engineering-toolkit/eslint-config/package.json:14]: History is written by the winners.

**Comment by FRSgit** [comment]
> I've added the resolutions field to the root package.json to allow local linting in this repo, due to its using the ESLint v9 and Prettier v2. The package itself is linted & formatted using itself.

I think you meant ESLint v8, right?

**Review by FRSgit — APPROVED** [review]
I think we can do a bit more in the docs, but other than that - great work!

File: engineering-toolkit/eslint-config/eslint.config.js (hunk 2 above)
Comment (grixu) [reply engineering-toolkit/eslint-config/eslint.config.js:5]: Ok, good to know 😄

Comment (grixu) [reply engineering-toolkit/eslint-config/eslint.config.js:5]: There is a `composer` or something like that, it's more sophisticated

**Pull request merged**